src/hocs/selection-hoc.jsx

```diff
@@ -13,23 +14,19 @@ const SelectionHOC = function (WrappedComponent) {
                 'removeItemById'
             ]);
         }
-        componentDidUpdate (prevProps) {
-            // Hovered item has changed
-            if ((this.props.hoveredItemId && this.props.hoveredItemId !== prevProps.hoveredItemId) ||
-                    (!this.props.hoveredItemId && prevProps.hoveredItemId)) {
-                // Remove the old hover item if any
-                this.removeItemById(prevProps.hoveredItemId);
+        componentDidUpdate () {
+            for (const itemId of this.props.removedItemIds) {
+                this.removeItemById(itemId);
             }
         }
         removeItemById (itemId) {
-            if (itemId) {
-                const match = paper.project.getItem({
-                    match: item => (item.id === itemId)
-                });
-                if (match) {
-                    match.remove();
-                }
+            const match = paper.project.getItem({
+                match: item => (item.id === itemId)
+            });
+            if (match) {
+                match.remove();
             }
+            this.props.clearRemovedItem(itemId);
```

Comment:
Should this call to `clearRemovedItem` be inside the `if (match)`? It seems like it would only matter if it's possible to have the same item ID twice (or more) in `removedItemIds`. Can that happen? If so, would it be more correct to call `clearRemovedItem` twice (or more), or just once, on that ID?